app/models/service_template_ansible_playbook.rb

```diff
@@ -11,63 +11,47 @@ def self.default_retirement_entry_point
   # options
   #   :name
   #   :description
-  #   :service_template_catalog
+  #   :service_template_catalog_id
   #   :config_info
   #     :provision
   #       :service_dialog_id (or)
   #       :new_dialog_name
   #       :variables
   #       :hosts
-  #       :credentials
+  #       :credential_id
+  #       :network_credential_id
+  #       :cloud_credential_id
   #       :playbook_id
   #     :retirement (same as provision)
   #     :reconfigure (same as provision)
   #
-  def self.create_catalog_item(options, _auth_user)
-    options      = options.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
+  def self.create_catalog_item(options, auth_user)
+    options      = options.deep_dup.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
     service_name = options[:name]
     description  = options[:description]
     config_info  = validate_config_info(options)
 
+    enhanced_config = config_info.deep_merge(create_job_templates(service_name, description, config_info, auth_user))
```

Comment:
Waiting for @syncrou's work be committed. But the spec test does not depend on it.